stackslib/src/net/download/nakamoto.rs

```diff
@@ -654,10 +688,10 @@ impl NakamotoTenureDownloader {
         &mut self,
         network: &mut PeerNetwork,
         neighbor_rpc: &mut NeighborRPC,
-    ) -> Result<Option<bool>, NetError> {
+    ) -> Result<bool, NetError> {
         if neighbor_rpc.has_inflight(&self.naddr) {
             test_debug!("Peer {} has an inflight request", &self.naddr);
-            return Ok(Some(true));
```

Comment:
I commented on this before this change -- but I think that this case actually does need to be handled differently. `Ok(Some(true))` will tell the caller that the downloader has made a request that its waiting for a response on, but that's not necessarily the case if the peer had some other request made to it, right?

Reply:
This has been reverted